tutorials/cloud_access/euclid-cloud-access.md

```diff
@@ -0,0 +1,303 @@
+def get_s3_fpath(cloud_access):
+    cloud_info = json.loads(cloud_access) # converts str to dict
+    bucket_name = cloud_info['aws']['bucket_name']
+    key = cloud_info['aws']['key']
+
+    return f'{bucket_name}/{key}'
+```
+
+```{code-cell} ipython3
+[get_s3_fpath(row['cloud_access']) for row in euclid_sci_img_tbl]
+```
+
+Let's also extract filter names to use when displaying the images:
+
+```{code-cell} ipython3
+def get_filter_name(instrument, bandpass):
+    return f'{instrument}_{bandpass}' if instrument!=bandpass else instrument
+```
+
+```{code-cell} ipython3
+[get_filter_name(row['instrument_name'], row['energy_bandpassname']) for row in euclid_sci_img_tbl]
+```
+
+## Retrieve image cutouts from the cloud
+These image files are very big (~1.4GB), so we use astropy's lazy-loading capability of FITS for better performance. (See [Obtaining subsets from cloud-hosted FITS files](https://docs.astropy.org/en/stable/io/fits/usage/cloud.html#fits-io-cloud).)
+
+```{code-cell} ipython3
+cutout_size = 1 * u.arcmin
+```
+
+```{code-cell} ipython3
+cutouts = []
+filters = []
+
+for row in euclid_sci_img_tbl:
+    s3_fpath = get_s3_fpath(row['cloud_access'])
+    filter_name = get_filter_name(row['instrument_name'], row['energy_bandpassname'])
+
+    with fits.open(f's3://{s3_fpath}', fsspec_kwargs={"anon": True}) as hdul:
+        print(f'Retrieving cutout for {filter_name} ...')
+        cutout = Cutout2D(hdul[0].section, 
+                          position=coord, 
+                          size=cutout_size, 
+                          wcs=WCS(hdul[0].header))
+        cutouts.append(cutout)
+        filters.append(filter_name)
+```
+
+```{code-cell} ipython3
+fig, axes = plt.subplots(2, 2, figsize=(4 * 2, 4 * 2), subplot_kw={'projection': cutouts[0].wcs})
+
+for idx, ax in enumerate(axes.flat):
+    norm = ImageNormalize(cutouts[idx].data, interval=PercentileInterval(99), stretch=AsinhStretch())
+    ax.imshow(cutouts[idx].data, cmap='gray', origin='lower', norm=norm)
+    ax.set_xlabel('RA')
+    ax.set_ylabel('Dec')
+    ax.text(0.95, 0.05, filters[idx], color='white', fontsize=14, transform=ax.transAxes, va='bottom', ha='right')
+
+plt.tight_layout()
+```
+
+## Find objects for the coordinates of our interest
+
++++
+
+### Locate MER catalogs in the bucket
+
+```{code-cell} ipython3
+s3.ls(f'{BUCKET_NAME}/q1/catalogs')
+```
+
+```{code-cell} ipython3
+s3.ls(f'{BUCKET_NAME}/q1/catalogs/MER_FINAL_CATALOG')
+```
+
+```{code-cell} ipython3
+s3.ls(f'{BUCKET_NAME}/q1/catalogs/MER_FINAL_CATALOG/102018211')
+```
+
+As per doc specification, we need `catalogs/MER_FINAL_CATALOG/{tile_id}/EUC_MER_FINAL-CAT*.fits` for listing the objects catalogued. But we only need to find objects in our coordinates of interest so we will use astroquery to do a spatial search in MER catalog (combined for all tiles).
+
++++
+
+### Get object IDs for the coordinates of our interest
+
+```{code-cell} ipython3
+tbl_catalogs = Irsa.list_catalogs(full=True).to_table()
```

Comment:
hmm, this should be a Table, too, I put it on my list

Reply:
This is now part of https://github.com/astropy/astroquery/pull/3252. I'm waiting on review on that one, but will merge it for the release.